sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,158 @@
+    }
+
+    private fun trimUrl(url: String): String {
+        // Remove any uuid from the url and replace it with a "*"
+        val trimmedUrl = url.replace(uuidRegex, "*")
+        if (URL(trimmedUrl).query == null) {
+            return trimmedUrl
+        }
+        // Remove any parameter from the url
+        return trimmedUrl.replace(URL(trimmedUrl).query, "").replace("?", "")
+    }
+
+    /**
+     * Sets the [Response] that will be sent in the breadcrumb [Hint].
+     * Also, it sets the protocol and status code in the breadcrumb and the call root span.
+     */
+    fun setResponse(response: Response) {
+        this.response = response
+        breadcrumb.setData("protocol", response.protocol.name)
+        breadcrumb.setData("status_code", response.code)
+        callRootSpan?.setData("protocol", response.protocol.name)
+        callRootSpan?.setData("status_code", response.code)
+        callRootSpan?.status = SpanStatus.fromHttpStatusCode(response.code)
+    }
+
+    fun setProtocol(protocolName: String?) {
+        if (protocolName != null) {
+            breadcrumb.setData("protocol", protocolName)
+            callRootSpan?.setData("protocol", protocolName)
+        }
+    }
+
+    fun setRequestBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("http.request_content_length", byteCount)
+            callRootSpan?.setData("http.request_content_length", byteCount)
+        }
+    }
+
+    fun setResponseBodySize(byteCount: Long) {
+        if (byteCount > -1) {
+            breadcrumb.setData("http.response_content_length", byteCount)
+            callRootSpan?.setData("http.response_content_length", byteCount)
+        }
+    }
+
+    /**
+     * Sets the success flag in the breadcrumb and the call root span to false.
+     * Also sets the [errorMessage] if not null.
+     */
+    fun setError(errorMessage: String?) {
+        breadcrumb.setData("success", false)
+        callRootSpan?.setData("success", false)
+        if (errorMessage != null) {
+            breadcrumb.setData("error_message", errorMessage)
+            callRootSpan?.setData("error_message", errorMessage)
+        }
+    }
+
+    /** Starts a span, if the callRootSpan is not null. */
+    fun startSpan(event: String) {
+        // Find the parent of the span being created. E.g. secureConnect is child of connect
+        val parentSpan = when (event) {
+            // PROXY_SELECT, DNS, CONNECT and CONNECTION are not children of one another
+            SECURE_CONNECT_EVENT -> eventSpans[CONNECT_EVENT]
+            REQUEST_HEADERS_EVENT -> eventSpans[CONNECTION_EVENT]
+            REQUEST_BODY_EVENT -> eventSpans[CONNECTION_EVENT]
+            RESPONSE_HEADERS_EVENT -> eventSpans[CONNECTION_EVENT]
+            RESPONSE_BODY_EVENT -> eventSpans[CONNECTION_EVENT]
+            else -> callRootSpan
+        } ?: callRootSpan
+        val span = parentSpan?.startChild("http.client", event) ?: return
```

Comment:
`m`: I think we should probably change the span op to something different (maybe `http.client.details`)?. I would assume we have a lot on the product side that is built specifically around `http.client` span op, and this might mess something up when these new spans do not adhere to the `http.client` spans structure (like method + url in the description, for example).

Reply:
makes sense 👍